packages/next/src/build/index.ts

```diff
@@ -1236,6 +1236,27 @@ export default async function build(
           appPaths = result.appPaths
           layoutPaths = result.layoutPaths
           // Note: defaultPaths are not used in the build process, only for slot detection in generating route types
+
+          // Copy static metadata files for production filesystem serving.
+          // Not using a loader so that it works on all bundlers.
+          if (result.staticMetadataFiles.size > 0) {
+            await nextBuildSpan
+              .traceChild('copy-static-metadata-files')
+              .traceAsyncFn(async () => {
+                for (const [
+                  requestPath,
+                  relativePath,
+                ] of result.staticMetadataFiles) {
+                  const outPath = path.join(
+                    distDir,
+                    'static/metadata',
```

Comment:
Why are these being output to `static/metadata` instead of server folder still? This would require builder change and also we would need to prevent immutable header being applied as these asset names don't include their content hash.